### PR TITLE
refactor: #158 remove no-explicit-any ignore eslint rule

### PR DESCRIPTION
**Pull request by miquelbeltran**
### Description :memo:

- Removes the `no-explicit-any` rule from the eslint configuration, applies code changes.
- Part of #158 
- Depends on #145 

**Type of change**

- [x] tech debt

**Updates**

- Removed ignored rule from config file.
- set specific type for `timerId` instead of `any`.
- improve `filterKeys` internal function.
- improve `setErrorDetails` public function, still allow `any` for end users.
- improve `IndexableError` type.
- improve worker callback type, even if the parameter is not used.
- allow `any` as `CustomData` so end users can provide whatever data they want.

### Test plan :test_tube:

- Run the `npm ci`, `npm test` and `npm run eslint` commands to ensure no issues raised.
- Verified that there were test cases using those changed functions.
- Run examples.

### Author to check :eyeglasses:
- [x] Project and all contained modules builds successfully
- [x] Self-/dev-tested 
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### Reviewer to check :heavy_check_mark:
- [ ] Project and all contained modules builds successfully
- [ ] Change has been dev-/reviewer-tested, where possible
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

**Comment by miquelbeltran** [comment]
PR is set to "into [145-eslint-setup](https://github.com/MindscapeHQ/raygun4node/tree/145-eslint-setup)" temporarily, the idea is to set to `develop` once #156 is merged

File: lib/raygun.sync.worker.ts [hunk 1]
```diff
@@ -2,16 +2,17 @@ import fs from "fs";
 
 import * as transport from "./raygun.transport";
 import { SendOptions, SendOptionsWithoutCB } from "./types";
+import {IncomingMessage} from "http";
```
Comment (sumitramanga) [inline lib/raygun.sync.worker.ts:5]: Move the import up as well - 
```suggestion
import { IncomingMessage } from "http";
```

File: examples/express-sample/package-lock.json
Comment (sumitramanga) [inline examples/express-sample/package-lock.json:1]: Are the changes in this file intentional ?

**Review by sumitramanga — COMMENTED** [review]
A couple of comments to look into before I approve!

File: examples/express-sample/package-lock.json
Comment (miquelbeltran) [reply examples/express-sample/package-lock.json:1]: Unfortunately, yes, seems that they were uncommited when switching to `eslint`.

**Review by sumitramanga — APPROVED** [review]
LGTM! 🧇

**Pull request merged**